integration/spark/src/test/scala/org/apache/carbondata/integration/spark/testsuite/complexType/TestCompactionComplexType.scala

```diff
@@ -47,6 +47,33 @@ class TestCompactionComplexType extends QueryTest with BeforeAndAfterAll {
     sql("DROP TABLE IF EXISTS compactComplex")
   }
 
+  test("complex issue") {
+    sql("drop table if exists complex1")
+    sql("create table complex1 (arr array<String>) stored as carbondata")
+    sql("insert into complex1 select array('as') union all " +
+        "select array('sd','df','gh') union all " +
+        "select array('rt','ew','rtyu','jk','sder') union all " +
+        "select array('ghsf','dbv','fg','ty') union all " +
+        "select array('hjsd','fggb','nhj','sd','asd')")
+
```

Comment:
Please add test scenario with data as null. array(null)

Reply:
This is WIP temp, cannot merge this poc code. Why review?